org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Parser.java

```diff
@@ -13759,6 +13846,19 @@ public Expression parseExpression(char[] source, int offset, int length, Compila
 	this.scanner.setSource(source);
 	this.scanner.resetTo(offset, offset + length - 1);
 	try {
+		if (stopAtError) {
+			int nextToken;
+			try {
+				nextToken = this.scanner.getNextToken();
+			} catch(InvalidInputException iie) {
+				return null;
+			}
```

Comment:
May be we can return NullLiteral

Reply:
I think the only case where we want to create NullLiteral is when nextToken == TokenNameEOF. For the other cases, it's very likely that we will have some other syntax errors already reported.